SConstruct

```diff
@@ -495,11 +495,12 @@ if env["use_precise_math_checks"]:
 if env.editor_build and env["engine_update_check"]:
     env.Append(CPPDEFINES=["ENGINE_UPDATE_CHECK_ENABLED"])
 
-if not env.File("#main/splash_editor.png").exists():
-    # Force disabling editor splash if missing.
-    env["no_editor_splash"] = True
-if env["no_editor_splash"]:
-    env.Append(CPPDEFINES=["NO_EDITOR_SPLASH"])
+if env.editor_build:
```

Comment:
Looks good, but I would refactor the previous check for the update checker under a common `if env.editor_build:`.